### ISSUE TITLE
BUG: Many-to-many relationship fail on update

### ISSUE THREAD

**Issue by gersur** [kepelrs/nestjs-prisma-crud#644]
![image](https://github.com/kepelrs/nestjs-prisma-crud/assets/1871841/b2998a67-8dfd-4113-bbaf-5d1eec00332a)
Let's just say that I have these relationship tables `certificate`----`job_certification`----`job`

Here is the payload PATCH request:
```json
{
  "job_certification": [
    {
      "id": 1
    }
  ],
}
```

But when I ran it, it'll error

```js
  177 // Checks that entity is accessible considering id and crudQuery restrictions.
  178 const entity = await this.findOne(id, fullOpts);
  179
→ 180 await repo.update({
        where: {
          id: 1
        },
        data: {
          job_certification: {
            create: [
              {
                certification_id: 1
              }
            ],
            connect: undefined,
            disconnect: [
              {
                id: undefined
              }
            ]
            ~~~~~~~~~~~~~~~~~
          },
....
Argument `disconnect`: Invalid value provided. Expected job_certificationWhereUniqueInput, provided (Object).
```

https://github.com/kepelrs/nestjs-prisma-crud/blob/47667e5851b5e037c81aa231fe2aec0ed18dc655/src/crud/helpers.ts#L224-L240

Then I tried to debug from function `getIdsToDisconnect`
```
param originalConnections [ { job_id: 1, certification_id: 1 } ]
param stillConnected []
param idPropertyName id
originalIds [ undefined ]
stillConnectedSet Set(0) {}
forDisconnecting [ { id: undefined } ]
```
So, because I executed in `job` service, it will use the job's `idPropertyName` which is `id`.
But the many-to-many table is not using `id` as the column name, it will error

I guess the solution is to be able override `idPropertyName` just for many-to-many relationship

**Comment by kepelrs** [comment]
Hi, thanks for bringing this up. I'm surprised I or someone else didn't hit this earlier.

I'll need to have a look at this soon. The low hanging fix would be indeed to somehow let the user specify the property name. But I'd be happier if we could infer it, maybe by it traversing some Prisma generated code, or something similar.

**Comment by kepelrs** [comment]
## TLDR

1. Add an `id` column to your explicit table
2. Make the `job_id` + `certification_id` combination unique in your prisma schema (but do not mark it as the primary id)
3. Make sure all your project's tables have an `id` column

## Full observations

I finally had a minute to look further into this and understand it better. Turns out many-to-many relations are actually supported and working as designed (see `./test/prisma/schema.prisma` file, Post <-> Category relation).

The simplest recommended way is to use [Implicit many-to-many relations](https://www.prisma.io/docs/orm/prisma-schema/data-model/relations/many-to-many-relations#implicit-many-to-many-relations), where you don't declare the middle table in your Prisma schema.

If you really need to use [Explicit many-to-many relations](https://www.prisma.io/docs/orm/prisma-schema/data-model/relations/many-to-many-relations#explicit-many-to-many-relations) by declaring the table in the Prisma schema (because for example you need extra columns on that table), then my current recommendation is:

1. Add an `id` column to your explicit table
2. Make the `job_id` + `certification_id` combination unique in your prisma schema (but do not mark it as the primary id)
3. Make sure all your project's tables have an `id` column

Some context as additional info for my future self and others who might interested:

-   While exploring enhancements to this, I disliked the option of adding a config option that allows specifying the relation's `idPropertyName`, as I feel like this loses coherence and introduces maintainability problems by forcing people to update and match these id's in multiple places.
-   Instead, we should refactor the base PrismaCrudService to be able to traverse a relation graph (using Prisma's existing APIs or some graph built by ourselves) and correctly infer the `idPropertyName` of the nested relation.
-   The point above has been suggested before and can have the benefit of allowing us to infer other useful stuff as well. But is larger work that will only happen if:
    1. I'm really free (not likely)
    2. This becomes a blocker for my own projects
    3. Some other helpful soul decides to contribute it (I can offer some assistance)
